nft-simple/src/nft_core.rs

```diff
@@ -133,40 +144,78 @@ impl NonFungibleTokenCore for Contract {
     }
 
     #[payable]
-    fn nft_approve_account_id(&mut self, token_id: TokenId, account_id: ValidAccountId) -> bool {
-        assert_one_yocto();
-        let mut token = self.tokens_by_id.get(&token_id).expect("Token not found");
-        assert_eq!(&env::predecessor_account_id(), &token.owner_id);
+    fn nft_approve(
+        &mut self,
+        token_id: TokenId,
+        account_id: ValidAccountId,
+        msg: Option<String>,
+    )  -> Option<Promise> {
```

Comment:
```suggestion
    )  {
```

Reply:
Okay, this is what I thought it would be. Talked to Chad yesterday about the conversation you three (Matt, too) had and he explained the reason for adding this to the NEP:

<img width="895" alt="Screen Shot 2021-03-26 at 1 16 24 PM" src="https://user-images.githubusercontent.com/1042667/112688093-9c340d80-8e35-11eb-91d0-54de5bf3b16a.png">

https://nomicon-pr-174.onrender.com/Standards/NonFungibleToken/ApprovalManagement.html#interface

Should we remove/modify the spec where it has:

    void|Promise<any>

Reply:
It will, but Rust near-sdk is a bit more complicated. So you have to do this differently.